backend/src/main/java/fk/prof/backend/BackendManager.java

```diff
@@ -123,6 +139,20 @@ public BackendManager(ConfigManager configManager) throws Exception {
     return result;
   }
 
+  private void initStorage() {
+    JsonObject s3Config = configManager.getS3Config();
+    ExecutorService storageExecSvc = new InstrumentedExecutorService(Executors.newFixedThreadPool(configManager.getStorageThreadPoolSize()),
+        SharedMetricRegistries.getOrCreate(ConfigManager.METRIC_REGISTRY), "executors.fixed_thread_pool.storage");
+    this.storage = new S3AsyncStorage(s3Config.getString("endpoint"), s3Config.getString("access.key"), s3Config.getString("secret.key"),
+        storageExecSvc);
+
+    JsonObject bufferPoolConfig = configManager.getFixedSizeBufferPool();
+    GenericObjectPoolConfig poolConfig = new GenericObjectPoolConfig();
+    poolConfig.setMaxTotal(bufferPoolConfig.getInteger("max.total"));
+    poolConfig.setMaxIdle(bufferPoolConfig.getInteger("max.idle"));
+
+    this.bufferPool = new GenericObjectPool<>(new ByteBufferPoolFactory(bufferPoolConfig.getInteger("buffer.size"), false), poolConfig);
```

Comment:
It was a quick solution for a buffer pool. I would wait for nfr to check for such contention because not all threads require this resource. Its only used by serialization/io thread.